### PR TITLE
fix: `ParseUser.save` fails when user is logged in

### PR DESCRIPTION
**Pull request by mbfakourii**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

### Issue Description
fix save user

Closes: #819

### Approach
Use the new method to send Json in ```fromJson```

### TODOs before merging

- [ ] Add tests
- [x] Add changes to documentation (guides, repository pages, in-code descriptions)
- [x] A changelog entry

### CODE REVIEW

**Comment by mtrezza** [comment]
Is this bug only affecting `ParseUser.save`, or does it also affect other objects? Could you give more details about this fix, so we can wrote a proper PR title?

**Comment by mbfakourii** [comment]
> Is this bug only affecting `ParseUser.save`, or does it also affect other objects? Could you give more details about the issue, so we can write a proper PR title / changelog entry?

This error is related to ```fromJson``` when server json values want to be converted to local model

File: packages/dart/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## [3.1.13](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-3.1.12...dart-3.1.13) (2023-02-13)
+
+### Bug Fixes
+
+* Bug for `user.save()` ([#819](https://github.com/parse-community/Parse-SDK-Flutter/issues/819))
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:5]:
```suggestion
## [3.1.13](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-3.1.12...dart-3.1.13) (2023-02-15)

### Bug Fixes

* `ParseUser.save` fails when user is logged in ([#819](https://github.com/parse-community/Parse-SDK-Flutter/issues/819))
```

**Review by mtrezza — APPROVED** [review]
Looks good!

**Pull request merged**